### DISCUSSION TITLE
Should I close the AppsAPIGroupDSL client?

### DISCUSSION THREAD

**Discussion by tinystorm** [fabric8io/kubernetes-client#6023] [Q&A] [answered]
I'm using a global client in my application. 
I notice that when I use the `KubernetesClient#apps` method, my IDE (IDEA) prompts me to use it within a try-with-resources block, indicating that I need to close this client. Should I simply ignore this prompt?

**Comment by manusa** [comment, marked as answer]
> Should I simply ignore this prompt?

Yes

> my IDE (IDEA) prompts me to use it within a try-with-resources block

It's sort of a false positive.
All clients are derived from your original client. You should close the "original" client once you've completed all the work that requires the client.
For example, in a Spring Boot, Quarkus, or Jakarta application, you'd normally declare a KubernetesClient bean that'd be reused throughout the lifecycle of the application.
You'd then close the client once the application shuts down.
